checks/evaluation/pinned_dependencies.go

```diff
@@ -158,13 +167,21 @@ func updatePinningResults(rr *checker.Dependency,
 }
 
 func generateText(rr *checker.Dependency) string {
-	if rr.Type == checker.DependencyUseTypeGHAction {
+	switch rr.Type {
+	case checker.DependencyUseTypeGHAction:
 		// Check if we are dealing with a GitHub action or a third-party one.
 		gitHubOwned := fileparser.IsGitHubOwnedAction(rr.Location.Snippet)
 		owner := generateOwnerToDisplay(gitHubOwned)
 		return fmt.Sprintf("%s %s not pinned by hash", owner, rr.Type)
+	case checker.DependencyUseTypeDockerfileContainerImage:
+		if rr.Name == nil {
+			break
+		}
+		if hash, err := crane.Digest(*rr.Name); err == nil { // if NO error
```

Comment:
The remediation text only needs to live in the remediation.go file. You have 2 options to make the remediation results part of the final JSON results:
1. Start updating the JSON structure we use in https://github.com/ossf/scorecard/blob/main/pkg/json.go, which is step towards achieving https://github.com/ossf/scorecard/issues/1874
2. Use the remediation results when constructing the final JSON string in  https://github.com/ossf/scorecard/blob/main/pkg/json.go, which is done in https://github.com/ossf/scorecard/blob/main/pkg/common.go#L30

As an example, you can see we do exactly that for SARIF output in https://github.com/ossf/scorecard/blob/main/pkg/sarif.go#L305-L308

For this PR, option 2 is the one I'd choose since option 1 is part of a bigger issue.

Reply:
Thanks for the feedback.

> The remediation text only needs to live in the remediation.go file

Sounds good. It felt weird having the logic in two places. I'll focus on `remediation.go` and revert the other commits.

> For this PR, option 2 is the one I'd choose since option 1 is part of a bigger issue.
I'll take a look at your provided links and take a stab at it